badger/attrcache.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 Pilosa Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package badger
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+func newCache(s *attrStore) *attrCache {
+	return &attrCache{
+		attrStore: s,
+		attrs:     make(map[uint64]map[string]interface{}),
+		mu:        sync.RWMutex{},
+	}
+}
+
+//attrCache wraps the attrstore and provides a cache
+type attrCache struct {
```

Comment:
I have considered this, I thought of restructuring things so that: 
`/attrStore` - stores the cache, nil implementation, as well as the tests that are in the main package. 
`/attrStore/Boltdb` - stores the boltdb implementation.
`/attrStore/Boltdb` - stores the badger implementation.

I also considered changing the `pilosa.AttrStore` interface to make it more friendly to the cache wrapper by returning the data that was updated in`SetAttrs` and `SetBulkAttrs`. Although I would rather discuss this. On one hand I don't see returning an additional pointer as a big problem, in addition there may be value in exposing this to the user- it's nice to know what actually changed... On the other hand if it's not needed then it may be better to avoid it.

Although I find this cleaner (except for changing the interface), it didn't seem essential. So I just made a small a step in this direction.